src/tyre.mli

```diff
@@ -191,6 +191,80 @@ val terminated_list : sep:_ t -> 'a t -> 'a list t
 val separated_list : sep:_ t -> 'a t -> 'a list t
 (** [separated_list ~sep tyre] is equivalent to [opt (e <&> list (sep *> e))]. *)
 
+module Charset: sig
+  (** {2 Sets of characters}
+    Sets of characters support more operations than regular regexps, as you can
+    diff them, so they have a specific type that allows these operations.
+
+    To convert to a regular [Tyre.t], use {!charset}. *)
+
+  type t
+  (** A set of characters. *)
+
+  val not: t -> t
+  val union: t -> t -> t
+  val inter: t list -> t
+  val diff: t -> t -> t
+  val compl : t list -> t
```

Comment:
Why is union binary but intersection n-ary ?

Also, what's the semantics of the n-ary complement ? Shouldn't it be unary instead ?

Reply:
inter is taken straight from Re, where it is n-ary. `union` I did myself, but you are right it should also be n-ary. Its implementation is `Re.alt`.

compl is the complement of the union. Its taken straight out of `Re`, I added `not` which is the unary version. Maybe compl should go away, but I am not sure how much the compatibility with `Re` is a goal